catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/PipelineRepository.java

```diff
@@ -323,18 +321,15 @@ public void entitySpecificUpdate() throws IOException {
     }
 
     private void updateTasks(Pipeline origPipeline, Pipeline updatedPipeline) throws JsonProcessingException {
-      // Airflow lineage backend gets executed per task in a DAG. This means we will not a get full picture of the
-      // pipeline in each call. Hence, we may create a pipeline and add a single task when one task finishes in a
-      // pipeline in the next task run we may have to update. To take care of this we will merge the tasks
-      List<Task> updatedTasks = Optional.ofNullable(updatedPipeline.getTasks()).orElse(Collections.emptyList());
-      List<Task> origTasks = Optional.ofNullable(origPipeline.getTasks()).orElse(Collections.emptyList());
-
-      // TODO this might not provide distinct
-      updatedTasks = Stream.concat(origTasks.stream(), updatedTasks.stream()).distinct().collect(Collectors.toList());
```

Comment:
we will not record any change with this approach right?
basically it will be empty based on the code here, unless I am missing something?

Reply:
If I understood the behaviour correctly, the `recordListChange` call will mark as deleted the current tasks not coming from the `updated` and as added those new elements that were not previously there. I replicated the approach with have for the `Columns`. Should we handle this with something more specific?